app/helpers/generic_file_helper.rb

```diff
@@ -21,8 +21,9 @@ def render_download_link text = nil
     link_to (text || "Download"), download_path(@generic_file), { id: "file_download", target: "_new", data: { label: @generic_file.id } }
   end
 
-  def download_path gf
-    GenericFile.find(gf.id).doi_url || sufia.download_path(gf.id)
+  # sufia.download path is from ActionDispatch::Routing::PolymorphicRoutes#polymorphic_path
+  def download_path gf, options = {}
+    GenericFile.find(gf.id).doi_url || sufia.download_path(gf.id, options)
```

Comment:
For 1 you might want to expand this logic and only do the find if it isn't a GenericFile but is a SolrDocument? I'm thinking mostly that ActiveFedora#find is sloooow and avoiding it when you can is good. 

For 2 I was thinking something alone the lines of:

```ruby
    gf.doi_url.present? ? gf.doi_url : sufia.download_path(gf.id, *more)
```

It looks like the linkable? call takes care of the empty string concern, though.

3) I guess my concern with the assumption is that we're shadowing the download resource route from Sufia here, and I guess my philosophy from a principle of least astonishment/liskov substitution principle point of view is that we shouldn't radically change the contract of that method; you can call Sufia's download_path with non-GenericFiles, but calling our shadowing function with a non-GenericFile will throw an exception

There's also a little tiny part of my that, however unlikely, can imagine someone calling download_path on some other type of object entirely that happens to respond to id with an id that coincidently matches a GenericFile's id, and then everything works entirely by accident in the worst possible way.

The format of ActiveFedora ids probably helps mitigate that, though